src/main/capi/table_description-c.cpp

```diff
@@ -0,0 +1,78 @@
+#include "duckdb/main/capi/capi_internal.hpp"
+
+using duckdb::Connection;
+using duckdb::ErrorData;
+using duckdb::TableDescription;
+using duckdb::TableDescriptionWrapper;
+
+// FIXME: should the table name be its own struct for extensibility?
+// i.e duckdb_table_path
+// duckdb_table_path_set_schema(...)
+// duckdb_table_path_set_catalog(...)
+duckdb_state duckdb_table_description_create(duckdb_connection connection, const char *schema, const char *table,
+                                             duckdb_table_description *out) {
+	Connection *conn = reinterpret_cast<Connection *>(connection);
+
+	if (!connection || !table || !out) {
+		return DuckDBError;
+	}
+	if (schema == nullptr) {
+		schema = DEFAULT_SCHEMA;
+	}
+	auto wrapper = new TableDescriptionWrapper();
+	*out = (duckdb_table_description)wrapper;
+	try {
+		wrapper->description = conn->TableInfo(schema, table);
+	} catch (std::exception &ex) {
+		ErrorData error(ex);
+		wrapper->error = error.RawMessage();
+		return DuckDBError;
+	} catch (...) { // LCOV_EXCL_START
+		wrapper->error = "Unknown Connection::TableInfo error";
+		return DuckDBError;
+	} // LCOV_EXCL_STOP
+	return DuckDBSuccess;
+}
+
+duckdb_state duckdb_table_description_destroy(duckdb_table_description *table) {
+	if (!table || !*table) {
+		return DuckDBError;
+	}
+	auto wrapper = reinterpret_cast<TableDescriptionWrapper *>(*table);
+	if (wrapper) {
+		delete wrapper;
+	}
+	*table = nullptr;
+	return DuckDBSuccess;
+}
+
+const char *duckdb_table_description_error(duckdb_table_description table) {
+	if (!table) {
+		return nullptr;
+	}
+	auto wrapper = reinterpret_cast<TableDescriptionWrapper *>(table);
+	if (wrapper->error.empty()) {
+		return nullptr;
+	}
+	return wrapper->error.c_str();
+}
+
+// other methods could be:
+// duckdb_column_is_generated(...)
+// duckdb_column_get_name(...)
+// duckdb_column_get_type(...)
+
+duckdb_state duckdb_column_has_default(duckdb_table_description table_description, idx_t index, bool *out) {
+	auto wrapper = reinterpret_cast<TableDescriptionWrapper *>(table_description);
+	if (!wrapper || !out) {
+		return DuckDBError;
+	}
+
+	auto &table = wrapper->description;
+	if (index >= table->columns.size()) {
+		return DuckDBError;
```

Comment:
Not sure I follow

Reply:
I think having the error field in any of our structs is a bad idea, I still really think we should make a `duckdb_error` struct and change the entire API to use these.

It only creates confusion as to when objects should be deleted, if I run a `create` method and that returns an error, the object should not have been created.

To be able to properly report an error in that scenario we would instead need to create an object so we have a place to store the error, which allocates memory that needs to be freed at some point.
So that would then mean `duckdb_table_description_destroy` should be called even if the call returns `DuckDBError`

Reply:
I looked at where I am using the `error` field, and I really don't agree with this (for the reasons mentioned above)
So I removed the error field entirely